New mypy error with asyncio.Task

```
❯ git checkout 67d875b785d4f79da7b76413ae62ad71a1259fc3
Note: switching to '67d875b785d4f79da7b76413ae62ad71a1259fc3'.
❯ python3.11 -m mypy pymodbus
Success: no issues found in 58 source files
❯ git checkout dev
Previous HEAD position was 67d875b7 Solve serial close raise problem.
Switched to branch 'dev'
Your branch is up to date with 'upstream/dev'.

❯ python3.11 -m mypy pymodbus
pymodbus/client/base.py:267: error: Statement is unreachable  [unreachable]
pymodbus/client/base.py:274: error: Incompatible types in assignment (expression has type "Task[Any]", variable has type "None")  [assignment]
Found 2 errors in 1 file (checked 60 source files)

❯ python3.8 -m mypy pymodbus
Success: no issues found in 60 source files
```

It appears that Python3.11 `mypy` catches this, but Python3.8 does not (which is why the CI didn't flag it).  My guess is that `asyncio` has improved its type definitions.  Anyways, it's an easy fix with a hint; I'll prepare a PR.

--- a/pymodbus/client/base.py
+++ b/pymodbus/client/base.py
@@ -119,7 +119,7 @@ class ModbusBaseClient(ModbusClientMixin, BaseTransport):
         self.state = ModbusTransactionState.IDLE
         self.last_frame_end: float = 0
         self.silent_interval: float = 0
-        self._reconnect_task = None
+        self._reconnect_task: asyncio.Task = None
 
         # Initialize  mixin
         ModbusClientMixin.__init__(self)

